lib/builders/BootstrapperBuilder.js

```diff
@@ -0,0 +1,309 @@
+/*
+ * catberry
+ *
+ * Copyright (c) 2015 Denis Rechkunov and project contributors.
+ *
+ * catberry's license follows:
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * This license applies to all parts of catberry that are not externally
+ * maintained libraries.
+ */
+
+'use strict';
+
+module.exports = BootstrapperBuilder;
+
+var path = require('path'),
+	util = require('util'),
+	pfs = require('../promises/fs'),
+	moduleHelper = require('../helpers/moduleHelper'),
+	requireHelper = require('../helpers/requireHelper');
+
+var BOOTSTRAPPER_FILENAME = 'Bootstrapper.js',
+	BROWSER_ROOT_PATH = path.join(__dirname, '..', '..', 'browser'),
+	STORE_FORMAT = '\n{name: \'%s\', constructor: %s}',
+	STORES_REPLACE = '/**__stores**/',
+	COMPONENT_FORMAT = '\n{' +
+		'name: \'%s\', ' +
+		'constructor: %s, ' +
+		'properties: %s, ' +
+		'templateSource: \'%s\', ' +
+		'errorTemplateSource: %s' +
```

Comment:
The module `BootstrapperBuilder` generates bootstrapper from a template and includes descriptors for every store and component. The result can look like this:
```javascript
var stores = [

{name: 'Pages', constructor: require('./catberry_stores/Pages.js')},
{name: 'static/Documentation', constructor: require('./catberry_stores/static/Documentation.js')},
{name: 'static/Overview', constructor: require('./catberry_stores/static/Overview.js')},
{name: 'static/Quotes', constructor: require('./catberry_stores/static/Quotes.js')}
];

var components = [

{name: 'bottom-scripts', constructor: require('./catberry_components/bottom-scripts/index.js'), properties: {"name":"bottom-scripts","template":"./template.hbs","errorTemplate":"./error.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "<script src=\\"/public/lib/highlight/highlight.pack.js\\"></script>\\n<script>\\n\twindow.hljs.configure({\\n\t\tlanguages: [\\n\t\t\t\'javascript\',\\n\t\t\t\'json\',\\n\t\t\t\'html\'\\n\t\t]\\n\t});\\n</script>\\n<script id=\\"twitter-wjs\\" src=\\"//platform.twitter.com/widgets.js\\"></script>\\n<cat-ga id=\\"analytics-tracking-code\\"></cat-ga>";\n  },"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "\\n";\n  },"useData":true}'},
{name: 'document', constructor: require('./catberry_components/document/index.js'), properties: {"name":"document","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var helper, functionType="function", helperMissing=helpers.helperMissing, escapeExpression=this.escapeExpression;\n  return "<!DOCTYPE html>\\n<html lang=\\""\n    + escapeExpression(((helper = (helper = helpers.locale || (depth0 != null ? depth0.locale : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"locale","hash":{},"data":data}) : helper)))\n    + "\\">\\n<head cat-store=\\"Pages\\"\\n\t  prefix=\\"og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#\\">\\n</head>\\n<body>\\n\t<cat-header id=\\"document-header\\"></cat-header>\\n\t<cat-pages id=\\"document-pages\\" cat-store=\\"Pages\\"></cat-pages>\\n\t<cat-footer id=\\"document-footer\\"></cat-footer>\\n\t<cat-bottom-scripts id=\\"document-bottom-scripts\\"></cat-bottom-scripts>\\n</body>\\n</html>\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'footer', constructor: require('./catberry_components/footer/index.js'), properties: {"name":"footer","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helper, functionType="function", helperMissing=helpers.helperMissing, buffer = "<footer>\\n\t<div class=\\"ui vertical inverted orange footer segment\\">\\n\t\t<div class=\\"ui stackable grid\\">\\n\t\t\t<div class=\\"one wide column\\"></div>\\n\t\t\t<div class=\\"three wide column fork-phrase\\">\\n\t\t\t\t<iframe src=\\"https://ghbtns.com/github-btn.html?user=catberry&repo=catberry-homepage&type=fork\\"\\n\t\t\t\t\t\tframeborder=\\"0\\" scrolling=\\"0\\" width=\\"54px\\" height=\\"20px\\"></iframe>\\n\t\t\t\t<span>this website</span>\\n\t\t\t</div>\\n\t\t\t<div class=\\"seven wide column\\"></div>\\n\t\t\t<div class=\\"three wide right aligned column\\">\\n\t\t\t\t<div class=\\"borderless title item\\">\\n\t\t\t\t\t";\n  stack1 = ((helper = (helper = helpers.copyrightText || (depth0 != null ? depth0.copyrightText : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"copyrightText","hash":{},"data":data}) : helper));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t</div>\\n\t</div>\\n</footer>\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'ga', constructor: require('./catberry_components/ga/index.js'), properties: {"name":"ga","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "<script>\\n\t(function(i,s,o,g,r,a,m){i[\'GoogleAnalyticsObject\']=r;i[r]=i[r]||function(){\\n\t\t(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),\\n\t\tm=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)\\n\t})(window,document,\'script\',\'//www.google-analytics.com/analytics.js\',\'ga\');\\n</script>";\n  },"useData":true}', errorTemplateSource: null},
{name: 'head', constructor: require('./catberry_components/head/index.js'), properties: {"name":"head","template":"./template.hbs","logic":"index.js"}, templateSource: '{"1":function(depth0,helpers,partials,data) {\n  var stack1, helperMissing=helpers.helperMissing, buffer = " / ";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, (depth0 != null ? depth0.pageTitleKey : depth0), {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer;\n},"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helper, helperMissing=helpers.helperMissing, escapeExpression=this.escapeExpression, functionType="function", buffer = "<meta charset=\\"utf-8\\">\\n<title>";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  stack1 = helpers[\'if\'].call(depth0, (depth0 != null ? depth0.pageTitleKey : depth0), {"name":"if","hash":{},"fn":this.program(1, data),"inverse":this.noop,"data":data});\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "</title>\\n<meta http-equiv=\\"x-ua-compatible\\" content=\\"edge\\">\\n<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">\\n<meta name=\\"description\\" content=\\""\n    + escapeExpression(((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "DESCRIPTION", {"name":"l10n","hash":{},"data":data})))\n    + "\\">\\n\\n<link rel=\\"stylesheet\\" href=\\"/public/lib/semantic-ui/semantic.min.css\\">\\n<link rel=\\"stylesheet\\" href=\\"/public/style.css\\">\\n<link rel=\\"stylesheet\\" href=\\"/public/lib/highlight/styles/github.css\\">\\n\\n<!-- Begin Twitter summary card-->\\n<meta name=\\"twitter:card\\" content=\\"summary\\" />\\n<meta name=\\"twitter:site\\" content=\\"@catberryjs\\" />\\n<meta name=\\"twitter:title\\" content=\\"";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  stack1 = helpers[\'if\'].call(depth0, (depth0 != null ? depth0.pageTitleKey : depth0), {"name":"if","hash":{},"fn":this.program(1, data),"inverse":this.noop,"data":data});\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\" />\\n<meta name=\\"twitter:description\\" content=\\""\n    + escapeExpression(((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "DESCRIPTION", {"name":"l10n","hash":{},"data":data})))\n    + "\\" />\\n<meta name=\\"twitter:image\\" content=\\""\n    + escapeExpression(((helper = (helper = helpers.socialLogo || (depth0 != null ? depth0.socialLogo : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"socialLogo","hash":{},"data":data}) : helper)))\n    + "\\" />\\n<meta name=\\"twitter:url\\" content=\\""\n    + escapeExpression(((helper = (helper = helpers.location || (depth0 != null ? depth0.location : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"location","hash":{},"data":data}) : helper)))\n    + "\\" />\\n<!-- End Twitter summary card-->\\n\\n<!-- Begin Open Graph markup -->\\n<meta property=\\"og:title\\" content=\\"";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  stack1 = helpers[\'if\'].call(depth0, (depth0 != null ? depth0.pageTitleKey : depth0), {"name":"if","hash":{},"fn":this.program(1, data),"inverse":this.noop,"data":data});\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\" />\\n<meta property=\\"og:description\\" content=\\""\n    + escapeExpression(((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "DESCRIPTION", {"name":"l10n","hash":{},"data":data})))\n    + "\\" />\\n<meta property=\\"og:type\\" content=\\"website\\" />\\n<meta property=\\"og:url\\" content=\\""\n    + escapeExpression(((helper = (helper = helpers.location || (depth0 != null ? depth0.location : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"location","hash":{},"data":data}) : helper)))\n    + "\\" />\\n<meta property=\\"og:image\\" content=\\""\n    + escapeExpression(((helper = (helper = helpers.socialLogo || (depth0 != null ? depth0.socialLogo : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"socialLogo","hash":{},"data":data}) : helper)))\n    + "\\" />\\n<!-- End Open Graph markup -->\\n\\n<script src=\\"/l10n.js\\"></script>\\n<script src=\\"/public/bundle.js\\" async></script>";\n},"useData":true}', errorTemplateSource: null},
{name: 'header', constructor: require('./catberry_components/header/index.js'), properties: {"name":"header","template":"./template.hbs","errorTemplate":"./error.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helperMissing=helpers.helperMissing, buffer = "<header>\\n\t<div class=\\"ui fixed inverted orange menu\\">\\n\t\t<div class=\\"borderless item\\"></div>\\n\t\t<div class=\\"borderless title item\\">\\n\t\t\t<a \thref=\\"/\\">";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "</a>\\n\t\t</div>\\n\t\t<div class=\\"borderless non-padded item\\">\\n\t\t\t<a href=\\"https://twitter.com/catberryjs\\" target=\\"_blank\\" title=\\"Twitter\\">\\n\t\t\t\t<i class=\\"twitter big white icon\\"></i>\\n\t\t\t</a>\\n\t\t</div>\\n\t\t<div class=\\"borderless non-padded item\\">\\n\t\t\t<a href=\\"https://github.com/catberry/catberry\\" target=\\"_blank\\" title=\\"Github Repository\\">\\n\t\t\t\t<i class=\\"github big white icon\\"></i>\\n\t\t\t</a>\\n\t\t</div>\\n\t\t<div class=\\"borderless non-padded item\\">\\n\t\t\t<a href=\\"https://gratipay.com/Catberry\\" target=\\"_blank\\" title=\\"Support us on Gratipay\\">\\n\t\t\t\t<i class=\\"gittip big white icon\\"></i>\\n\t\t\t</a>\\n\t\t</div>\\n\t\t<div class=\\"right menu\\">\\n\t\t\t<div class=\\"social\\">\\n\t\t\t\t<a href=\\"https://twitter.com/share\\" class=\\"twitter-share-button\\"\\n\t\t\t\t   data-via=\\"catberryjs\\" data-related=\\"catberryjs\\" data-hashtags=\\"javascript\\" data-lang=\\"en\\">Tweet</a>\\n\t\t\t\t<iframe src=\\"http://ghbtns.com/github-btn.html?user=catberry&repo=catberry&type=watch&count=true\\"\\n\t\t\t\t\t\theight=\\"20\\" width=\\"118\\" frameborder=\\"0\\" scrolling=\\"0\\"\\n\t\t\t\t\t\tstyle=\\"width:118px; height: 20px;\\" allowTransparency=\\"true\\"></iframe>\\n\t\t\t</div>\\n\t\t</div>\\n\t\t<div class=\\"right menu\\">\\n\t\t\t<cat-section-navigation id=\\"header-section-navigation\\" cat-store=\\"Pages\\">\\n\t\t\t</cat-section-navigation>\\n\t\t</div>\\n\t\t<div class=\\"borderless item\\"></div>\\n\t</div>\\n</header>\\n";\n},"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "\\n";\n  },"useData":true}'},
{name: 'page-documentation', constructor: require('./catberry_components/page-documentation/index.js'), properties: {"name":"page-documentation","template":"./template.hbs","errorTemplate":"./error.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "<div class=\\"ui active loader\\"></div>\\n<cat-section-documentation id=\\"docs-section-docs\\" cat-store=\\"static/Documentation\\">\\n</cat-section-documentation>\\n";\n  },"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "\\n";\n  },"useData":true}'},
{name: 'page-main', constructor: require('./catberry_components/page-main/index.js'), properties: {"name":"page-main","template":"./template.hbs","errorTemplate":"./error.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "<div class=\\"ui active loader\\"></div>\\n<cat-section-card id=\\"main-section-card\\"></cat-section-card>\\n<div class=\\"ui divider\\"></div>\\n<cat-section-features id=\\"main-section-features\\"></cat-section-features>\\n<div class=\\"ui divider\\"></div>\\n<cat-section-quotes id=\\"main-section-quotes\\" cat-store=\\"static/Quotes\\"></cat-section-quotes>\\n";\n  },"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "\\n";\n  },"useData":true}'},
{name: 'page-overview', constructor: require('./catberry_components/page-overview/index.js'), properties: {"name":"page-overview","template":"./template.hbs","errorTemplate":"./error.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "<div class=\\"ui active loader\\"></div>\\n<cat-section-overview id=\\"overview-section-overview\\" cat-store=\\"static/Overview\\"></cat-section-overview>\\n";\n  },"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "\\n";\n  },"useData":true}'},
{name: 'pages', constructor: require('./catberry_components/pages/index.js'), properties: {"name":"pages","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var helper, functionType="function", helperMissing=helpers.helperMissing, escapeExpression=this.escapeExpression;\n  return "<section class=\\"catberry section\\">\\n\t<div class=\\"ui basic placeholder segment\\">\\n\t\t"\n    + escapeExpression(((helper = (helper = helpers.message || (depth0 != null ? depth0.message : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"message","hash":{},"data":data}) : helper)))\n    + "???\\n\t\t<cat-page-"\n    + escapeExpression(((helper = (helper = helpers.currentPage || (depth0 != null ? depth0.currentPage : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"currentPage","hash":{},"data":data}) : helper)))\n    + " id=\\"pages-current\\"></cat-page-"\n    + escapeExpression(((helper = (helper = helpers.currentPage || (depth0 != null ? depth0.currentPage : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"currentPage","hash":{},"data":data}) : helper)))\n    + ">\\n\t</div>\\n</section>\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'section-card', constructor: require('./catberry_components/section-card/index.js'), properties: {"name":"section-card","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helperMissing=helpers.helperMissing, buffer = "<section class=\\"catberry padded section\\">\\n\t<div class=\\"ui page grid\\">\\n\t\t<div class=\\"nine wide column\\">\\n\t\t\t<div class=\\"ui secondary segment\\">\\n\t\t\t\t<div class=\\"ui huge header\\">\\n\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "CARD_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t<a href=\\"https://www.npmjs.com/packages/catberry\\" target=\\"_blank\\">\\n\t\t\t\t\t\t<img src=\\"/public/assets/section-card/images/npm-logo.svg\\" height=\\"16px\\">\\n\t\t\t\t\t</a>\\n\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "CARD_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t\t<p>\\n\t\t\t\t<div class=\\"ui small header\\">\\n\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "CARD_TO_INSTALL", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t</div>\\n\t\t\t\t<code class=\\"code hljs bash\\">\\n\t\t\t\t\t# npm install -g catberry-cli\\n\t\t\t\t</code>\\n\t\t\t\t</p>\\n\t\t\t\t<p>\\n\t\t\t\t<div class=\\"ui small header\\">\\n\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "CARD_TO_CREATE_EMPTY", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t</div>\\n\t\t\t\t<code class=\\"code hljs bash\\">\\n\t\t\t\t\t$ catberry init empty-handlebars\\n\t\t\t\t</code>\\n\t\t\t\t</p>\\n\t\t\t\t<p>\\n\t\t\t\t<div class=\\"ui small header\\">\\n\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "CARD_TO_CREATE_EXAMPLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t</div>\\n\t\t\t\t<code class=\\"code hljs bash\\">\\n\t\t\t\t\t$ catberry init example\\n\t\t\t\t</code>\\n\t\t\t\t</p>\\n\t\t\t\t<p>\\n\t\t\t\t<div class=\\"ui small header\\">\\n\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "CARD_TO_LIST_TEMPLATES", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\n\t\t\t\t</div>\\n\t\t\t\t<code class=\\"code hljs bash\\">\\n\t\t\t\t\t$ catberry init ?\\n\t\t\t\t</code>\\n\t\t\t\t</p>\\n\t\t\t</div>\\n\t\t</div>\\n\t\t<div class=\\"seven wide left aligned column\\">\\n\t\t\t<img class=\\"ui middle aligned image\\" src=\\"/public/assets/section-card/images/catberryjs.svg\\">\\n\t\t</div>\\n\t</div>\\n</section>\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'section-documentation', constructor: require('./catberry_components/section-documentation/index.js'), properties: {"name":"section-documentation","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helper, functionType="function", helperMissing=helpers.helperMissing, buffer = "<section class=\\"catberry padded markdown section doc\\">\\n\t";\n  stack1 = ((helper = (helper = helpers.html || (depth0 != null ? depth0.html : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"html","hash":{},"data":data}) : helper));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\n</section>\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'section-features', constructor: require('./catberry_components/section-features/index.js'), properties: {"name":"section-features","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helperMissing=helpers.helperMissing, buffer = "<section class=\\"catberry padded section\\">\\n\t<div class=\\"ui page grid\\">\\n\t\t<div class=\\"center aligned row\\">\\n\t\t\t<div class=\\"five wide column\\">\\n\t\t\t\t<div class=\\"ui icon center header\\">\\n\t\t\t\t\t<i class=\\"circular code icon\\"></i>\\n\t\t\t\t\t<div class=\\"content\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_ISOMORPHIC_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_ISOMORPHIC_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t</div>\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t\t<div class=\\"five wide column\\">\\n\t\t\t\t<div class=\\"ui icon header\\">\\n\t\t\t\t\t<i class=\\"circular puzzle icon\\"></i>\\n\t\t\t\t\t<div class=\\"content\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_COMPONENTS_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_COMPONENTS_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t</div>\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t\t<div class=\\"five wide column\\">\\n\t\t\t\t<div class=\\"ui icon header\\">\\n\t\t\t\t\t<i class=\\"circular exchange icon\\"></i>\\n\t\t\t\t\t<div class=\\"content\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_FLUX_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_FLUX_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t</div>\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t</div>\\n\\n\t\t<div class=\\"center aligned row\\">\\n\t\t\t<div class=\\"five wide column\\">\\n\t\t\t\t<div class=\\"ui icon header\\">\\n\t\t\t\t\t<i class=\\"circular unhide icon\\"></i>\\n\t\t\t\t\t<div class=\\"content\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_SEO_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_SEO_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t</div>\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t\t<div class=\\"five wide column\\">\\n\t\t\t\t<div class=\\"ui icon header\\">\\n\t\t\t\t\t<i class=\\"circular rocket icon\\"></i>\\n\t\t\t\t\t<div class=\\"content\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_FAST_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_FAST_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t</div>\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t\t<div class=\\"five wide column\\">\\n\t\t\t\t<div class=\\"ui icon header\\">\\n\t\t\t\t\t<i class=\\"circular checkmark box icon\\"></i>\\n\t\t\t\t\t<div class=\\"content\\">\\n\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_READY_TITLE", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\n\t\t\t\t\t\t<div class=\\"sub header\\">\\n\t\t\t\t\t\t\t";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, "FEATURE_READY_DESC", {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\n\t\t\t\t\t\t</div>\\n\t\t\t\t\t</div>\\n\t\t\t\t</div>\\n\t\t\t</div>\\n\t\t</div>\\n\t</div>\\n</section>\\n\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'section-navigation', constructor: require('./catberry_components/section-navigation/index.js'), properties: {"name":"section-navigation","logic":"./index.js","template":"./template.hbs","errorTemplate":"./error.hbs"}, templateSource: '{"1":function(depth0,helpers,partials,data) {\n  var stack1, helper, functionType="function", helperMissing=helpers.helperMissing, escapeExpression=this.escapeExpression, buffer = "\t\t<a class=\\"item";\n  stack1 = helpers[\'if\'].call(depth0, (depth0 != null ? depth0.isActive : depth0), {"name":"if","hash":{},"fn":this.program(2, data),"inverse":this.noop,"data":data});\n  if (stack1 != null) { buffer += stack1; }\n  buffer += "\\" href=\\"/"\n    + escapeExpression(((helper = (helper = helpers.name || (depth0 != null ? depth0.name : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"name","hash":{},"data":data}) : helper)))\n    + "\\">!!"\n    + escapeExpression(((helper = (helper = helpers.message || (depth0 != null ? depth0.message : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"message","hash":{},"data":data}) : helper)))\n    + " ";\n  stack1 = ((helpers.l10n || (depth0 && depth0.l10n) || helperMissing).call(depth0, (depth0 != null ? depth0.pageTitleKey : depth0), {"name":"l10n","hash":{},"data":data}));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "</a>\\n";\n},"2":function(depth0,helpers,partials,data) {\n  return " active";\n  },"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, buffer = "<nav>\\n";\n  stack1 = helpers.each.call(depth0, (depth0 != null ? depth0.pages : depth0), {"name":"each","hash":{},"fn":this.program(1, data),"inverse":this.noop,"data":data});\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "</nav>";\n},"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "error\\n";\n  },"useData":true}'},
{name: 'section-overview', constructor: require('./catberry_components/section-overview/index.js'), properties: {"name":"section-overview","template":"./template.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helper, functionType="function", helperMissing=helpers.helperMissing, buffer = "<section class=\\"catberry padded markdown section doc\\">\\n\t";\n  stack1 = ((helper = (helper = helpers.html || (depth0 != null ? depth0.html : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"html","hash":{},"data":data}) : helper));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\n</section>\\n";\n},"useData":true}', errorTemplateSource: null},
{name: 'section-quotes', constructor: require('./catberry_components/section-quotes/index.js'), properties: {"name":"section-quotes","template":"./template.hbs","errorTemplate":"./error.hbs","logic":"index.js"}, templateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  var stack1, helper, functionType="function", helperMissing=helpers.helperMissing, buffer = "<section class=\\"catberry padded section\\">\\n\t";\n  stack1 = ((helper = (helper = helpers.html || (depth0 != null ? depth0.html : depth0)) != null ? helper : helperMissing),(typeof helper === functionType ? helper.call(depth0, {"name":"html","hash":{},"data":data}) : helper));\n  if (stack1 != null) { buffer += stack1; }\n  return buffer + "\\n</section>\\n";\n},"useData":true}', errorTemplateSource: '{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {\n  return "\\n";\n  },"useData":true}'}
];
```